sdk/core/Azure.Core/src/Pipeline/HttpPipeline.cs

```diff
@@ -70,7 +70,9 @@ public ValueTask SendAsync(HttpMessage message, CancellationToken cancellationTo
         {
             message.CancellationToken = cancellationToken;
             AddHttpMessageProperties(message);
-            return _pipeline.Span[0].ProcessAsync(message, _pipeline.Slice(1));
+            var value = _pipeline.Span[0].ProcessAsync(message, _pipeline.Slice(1));
+            message.Response.EvaluateError(message);
```

Comment:
One unfortunate downside of this model is that policies won't see a correct IsError value.

Reply:
Would it work to put it in the transport policy and evaluate it when the Response is created?  I guess that would mean any user-specified transport would have to know to do this.

Reply:
Updated with a policy added to the pipeline that calls Response.EvaluateError()